### DISCUSSION TITLE
tls handshake problem

### DISCUSSION THREAD

**Discussion by DHMG-Git** [saran-surya/email_auth_node#69] [Q&A] [answered]
Hi,

I followed your guide and set up my own server. The service seems to be running fine and I made it accassable from the outside, but when I start my app that uses the email_auth package I get this console warning:

> I/flutter ( 6823): --- Package Error ---
> I/flutter ( 6823): HandshakeException: Handshake error in client (OS Error:
> I/flutter ( 6823): 	WRONG_VERSION_NUMBER(tls_record.cc:242))
> I/flutter ( 6823): --- End Package Error ---
> I/flutter ( 6823):
> I/flutter ( 6823): --- package Error ---
> I/flutter ( 6823): email-auth >> The remote server is not a valid.
> I/flutter ( 6823): email-auth >> configured server : "https://dhmg.ddns.net:5000"
> I/flutter ( 6823): --- package Error ---

Since it seems to be ssl/tls related I tried accessing the service with http instead. And indeed, the error changed to:

> I/flutter ( 6823): email-auth >> Initialising Email-Auth server
> I/flutter ( 6823):
> I/flutter ( 6823): --- package Error ---
> I/flutter ( 6823): email-auth >> The remote server is not a valid.
> I/flutter ( 6823): email-auth >> configured server : "http://dhmg.ddns.net:5000"
> I/flutter ( 6823): --- package Error ---

Since you never mentioned in your guide that I can provide a private key / path to a ssl certificate I assume your service always runs without ssl? If so, can you help me figure out why my remote server is not valid?

**Comment by saran-surya** [comment, marked as answer]
So regarding the server part, you are required to configure the domain. I still prefer using a service that has all the elements already configured. Probably Heroku does the job for me, right now trying to integrate with GCP as to start the test servers back again
